src/main/java/tech/blacksource/blacknectar/service/json/OperationResultJson.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * 
+ * Copyright 2017 BlackSource, LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package tech.blacksource.blacknectar.service.json;
+
+import java.util.Objects;
+
+import com.google.gson.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import tech.sirwellington.alchemy.annotations.arguments.Required;
+
+import static tech.blacksource.blacknectar.service.BlackNectarAssertions.hasField;
+import static tech.sirwellington.alchemy.arguments.Arguments.*;
+import static tech.sirwellington.alchemy.arguments.assertions.Assertions.notNull;
+import static tech.sirwellington.alchemy.arguments.assertions.StringAssertions.nonEmptyString;
+
+/**
+ * @author SirWellington
+ */
+public final class OperationResultJson implements AsJson
```

Comment:
There can definitely be a better name for this:

Maybe just simply `OperationResult` would suffice.